railties/lib/rails/tasks/routes.rake

```diff
@@ -1,7 +1,19 @@
+require 'optparse'
+
 desc 'Print out all defined routes in match order, with names. Target specific controller with CONTROLLER=x.'
 task routes: :environment do
   all_routes = Rails.application.routes.routes
   require 'action_dispatch/routing/inspector'
   inspector = ActionDispatch::Routing::RoutesInspector.new(all_routes)
-  puts inspector.format(ActionDispatch::Routing::ConsoleFormatter.new, ENV['CONTROLLER'])
+  options = { filter: ENV['CONTROLLER'] }
+
+  OptionParser.new do |opts|
+    opts.banner = "Usage: rake routes [options]"
+    opts.on("-g", "--grep PATTERN", String) do |pattern|
+      options[:pattern] = pattern
+    end
+  end.parse!(ARGV.reject { |x| x == "routes" })
+
+  puts inspector.format(ActionDispatch::Routing::ConsoleFormatter.new, options)
+  exit 0
```

Comment:
`exit` at the end will make sure that the extra arguments won't be interpreted as Rake task.